src/main/java/frc/robot/subsystems/Limelight.java

```diff
@@ -0,0 +1,166 @@
+    tySubscriber = NetworkTableInstance.getDefault().getDoubleTopic(fmtPath("ty")).subscribe(0.0);
+    taSubscriber = NetworkTableInstance.getDefault().getDoubleTopic(fmtPath("ta")).subscribe(0.0);
+    DoubleTopic pipelineTopic =
+        NetworkTableInstance.getDefault().getDoubleTopic(fmtPath("pipeline"));
+    this.pipelineSubcriber = pipelineTopic.subscribe(0.0);
+    this.pipelinePublisher = pipelineTopic.publish();
+    double[] defaultBotpos = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
+    botposSubscriber =
+        NetworkTableInstance.getDefault()
+            .getDoubleArrayTopic(fmtPath("botpose"))
+            .subscribe(defaultBotpos);
+  }
+
+  /* now its time for getter method chaingun, which I have to write manually because VS Code */
+  /* nvm im making a python script do this for me */
+
+  /**
+   * Returns whether the Limelight has a valid target.
+   *
+   * @return Whether or not the Limelight has a valid target. True if it does, false if it doesn't.
+   */
+  public boolean hasValidTarget() {
+    return tvSubscriber.get() == 1.0;
+  }
+
+  /**
+   * Gets the horizontal offset from the crosshair to the currently active target.
+   *
+   * @return tx; the horizontal offset from the crosshair to the target.
+   */
+  public double getTX() {
+    return txSubscriber.get();
+  }
+
+  /**
+   * Gets the vertical offset from the crosshair to the currently active target.
+   *
+   * @return ty; the vertical offset from the crosshair to the target.
+   */
+  public double getTY() {
+    return tySubscriber.get();
+  }
+
+  /**
+   * Gets the area of the target, 0% to 100% of the image.
+   *
+   * @return ta; the area of the target.
+   */
+  public double getTA() {
+    return taSubscriber.get();
+  }
+
+  /**
+   * Returns the robot's current pose relative to the field.
+   *
+   * <p>Units are meters; 0,0 is at the center of the field.
+   *
+   * <p>Will only work if the Limelight can see an AprilTag and read it correctly.
+   *
+   * <p>Notably, the Z, roll, and pitch values will always be zero, since we snap the bot to the
+   * floor on the Limelight.
+   *
+   * @return A {@link Pose3d}; the robot's current position relative to the field.
+   */
+  public Pose3d getBotpose() {
+    double[] botpos = botposSubscriber.get();
+    return new Pose3d(
+        botpos[0], botpos[1], botpos[2], new Rotation3d(botpos[3], botpos[4], botpos[5]));
+  }
+
+  /**
+   * Gets the robot's current pose relative to the field.
+   *
+   * <p>Units are meters; 0,0 is at the center of the field.
+   *
+   * <p>Will only work if the Limelight can see an AprilTag and read it correctly.
+   *
+   * @return A {@link Pose2d}; the robot's current position relative to the field in two dimensions.
+   */
+  public Pose2d getBotpose2d() {
+    double[] botpos = botposSubscriber.get();
+    return new Pose2d(botpos[0], botpos[1], Rotation2d.fromDegrees(botpos[5]));
+  }
+
+  /**
+   * Returns the current limelight pipeline.
```

Comment:
Limelight pipeline number

Also you need to go through and capitalize Limelight in all JavaDoc